db.go

```diff
@@ -1129,7 +1129,7 @@ func (m *meta) validate() error {
 		return ErrInvalid
 	} else if m.version != version {
 		return ErrVersionMismatch
-	} else if m.checksum != 0 && m.checksum != m.sum64() {
```

Comment:
Why this code is 'unreachable', i.e. why it's guaranteed that no file has m.checksum == 0?

Reply:
`m.checksum` might be `0`, but at that case `m.sum64()` is `0` too, so the check of `m.checksum!=0` is unnecessary

Reply:
What I mean is that the logic says: 'if page==0' than the checksum is not being verfied. 
It's not what should happen in the wild - especially using the current version of bbolt (might be property of storegev1). 

I see 2 theorethical scenarios:
  1. Someone did a file surgary using a tool that cannot compute checksum - phishy. 
  2. Someone is opening a really old file using current version of bbolot